src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -448,17 +466,28 @@ def get_webhook_data(
 def send_resource_change_webhook(
     installation_id: int, event: str, data: dict[str, Any], *args: Any, **kwargs: Any
 ) -> None:
-    installation = app_service.installation_by_id(id=installation_id)
-    if not installation:
-        logger.info(
-            "send_resource_change_webhook.missing_installation",
-            extra={"installation_id": installation_id, "event": event},
-        )
-        return
+    with SentryAppInteractionEvent(
+        operation_type=SentryAppInteractionType.SEND_WEBHOOK, event_type=event
+    ).capture() as lifecycle:
+        installation = app_service.installation_by_id(id=installation_id)
+        if not installation:
+            logger.info(
+                "send_resource_change_webhook.missing_installation",
+                extra={"installation_id": installation_id, "event": event},
+            )
+            return
 
-    send_webhooks(installation, event, data=data)
+        try:
+            send_webhooks(installation, event, data=data)
+        except SentryAppSentryError as e:
+            sentry_sdk.capture_exception(e)
+            lifecycle.record_failure(e)
+            return None
+        except (ApiHostError, ApiTimeoutError, RequestException, ClientError) as e:
+            lifecycle.record_halt(e)
```

Comment:
`ApiHostError` & `ApiTimeoutError` are raised from `send_and_save_webhook_request` when the [response is 503 or 504](https://github.com/getsentry/sentry/blob/crl/sa-slos-context-manager/src/sentry/utils/sentry_apps/webhooks.py#L189-L193). 

`RequestExceptions` occur when we get a `Timeout` or `ConnectionError`, these I also woudn't consider our server being at fault.

`ClientErrors `are anything <500 and also wouldn't be our fault, because it's the 3p responsibility to properly consume our requests.

We re raise since the `TASK_OPTIONS` already have the retry logic specified (i.e which errors to retry or ignore)

Reply:
JFYI that we will not encounter this block for NOT published apps (e.g internal & unpublished). As we only propagate up errors from `send_and_save_webhook_request` for published apps.